pasqal-cloud/pasqal_cloud/utils/mock_server.py

```diff
@@ -0,0 +1,67 @@
+import abc
+from typing import Any
+
+import requests_mock
+
+from pasqal_cloud.endpoints import Endpoints
+
+
+class BaseMockServer(abc.ABC):
+    """
+    A mock server used as a replacement for the Pasqal HTTP server during testing.
+
+    Intended use:
+    ```py
+    class MyMockServer(BaseMockServer):
+        ...
+
+    with MyMockServer():
```

Comment:
I like this interface! 

Currently the class is lacking enter an exit methods to allow the usage as a the context manager using the `with` keyword